app/src/main/java/org/wikipedia/descriptions/DescriptionEditView.java

```diff
@@ -109,32 +122,81 @@ public void editTaskEnabled(boolean enabled) {
 
     private void setHintText() {
         pageDescriptionLayout.setHintTextAppearance(R.style.DescriptionEditViewHintTextStyle);
-        pageDescriptionLayout.setHint(String.format(getContext().getString(R.string.description_edit_text_hint_per_language),
-                WikipediaApp.getInstance().language().getAppLanguageCanonicalName(pageTitle.getWikiSite().languageCode())));
+        pageDescriptionLayout.setHint(getHintText(pageTitle.getWikiSite().languageCode()));
+    }
+
+    private int getHeaderTextRes(boolean inReview) {
+        if (TextUtils.isEmpty(originalDescription)) {
+            if (inReview) {
+                if (invokeSource.name().contains(INVOKE_SOURCE_KEYWORD_CAPTION)) {
+                    return R.string.suggested_edits_review_image_caption;
+                } else {
+                    return R.string.suggested_edits_review_description;
+                }
+            } else {
+                if (invokeSource == SUGGESTED_EDITS_TRANSLATE_DESC || invokeSource == FEED_CARD_SUGGESTED_EDITS_TRANSLATE_DESC) {
+                    return R.string.description_edit_translate_description;
+                } else if (invokeSource == SUGGESTED_EDITS_ADD_CAPTION) {
+                    return R.string.description_edit_add_image_caption;
+                } else if (invokeSource == SUGGESTED_EDITS_TRANSLATE_CAPTION) {
+                    return R.string.description_edit_translate_image_caption;
+                } else {
+                    return R.string.description_edit_add_description_v2;
+                }
+            }
+        } else {
+            return R.string.description_edit_edit_description;
+        }
+    }
+
+    private CharSequence getLabelText(@NonNull String lang) {
+        if (invokeSource == SUGGESTED_EDITS_TRANSLATE_DESC || invokeSource == FEED_CARD_SUGGESTED_EDITS_TRANSLATE_DESC) {
+            return String.format(getContext().getString(R.string.description_edit_text_hint_per_language),
+                    WikipediaApp.getInstance().language().getAppLanguageCanonicalName(lang));
```

Comment:
The `getString()` function itself lets you add parameters into it. You don't need to use `String.format()`.

Reply:
Thanks, done!